reports-scheduler/src/test/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportInstanceIT.kt

```diff
@@ -0,0 +1,290 @@
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+
+        val reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+        val onDemandRequest = """
+            {}
+        """.trimIndent()
+        val onDemandResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/on_demand/$reportDefinitionId",
+            onDemandRequest,
+            200
+        )
+        val reportInstance = onDemandResponse.get("reportInstance").asJsonObject
+        val reportInstanceId = reportInstance.get("id").asString
+
+        val addReportDefinition = """
+            {
+                "reportDefinition":{
+                    "name":"new report definition",
+                    "isEnabled":true,
+                    "source":{
+                        "description":"new description",
+                        "type":"Dashboard",
+                        "origin":"localhost:5601",
+                        "id":"id"
+                    },
+                    "format":{
+                        "duration":"PT1H",
+                        "fileFormat":"Pdf",
+                        "limit":1000,
+                        "header":"optional header",
+                        "footer":"optional footer"
+                    },
+                    "trigger":{
+                        "triggerType":"OnDemand"
+                    },
+                    "delivery":{
+                        "recipients":["nobody@email.com"],
+                        "deliveryFormat":"LinkOnly",
+                        "title":"title",
+                        "textDescription":"textDescription",
+                        "htmlDescription":"optional htmlDescription",
+                        "channelIds":["optional_channelIds"]
+                    }
+                }
+            }
+            """.trimIndent()
+        val addReportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            addReportDefinition,
+            RestStatus.OK.status
+        )
+        val newReportDefinitionId = addReportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", newReportDefinitionId)
+
+        val newOnDemandResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/on_demand/$newReportDefinitionId",
+            onDemandRequest,
+            200
+        )
+        val newReportInstance = newOnDemandResponse.get("reportInstance").asJsonObject
+        val newReportInstanceId = newReportInstance.get("id").asString
+        Thread.sleep(3000)
```

Comment:
Does 1000 work? i had 3000 before but it seems longer than needed

Reply:
I'll check with 1000, this was left over from when I was debugging earlier